experimental/LieAlgebras/src/LieAlgebra.jl

```diff
@@ -195,19 +195,20 @@ end
 
 ###############################################################################
 #
-#   Attribute accessors
+#   Isomorphism to GAP accessors
 #
 ###############################################################################
 
-function _gap_object(L::LieAlgebra{C}) where {C<:RingElement}
-  # later change to storing an isomorphism instead
-  get_attribute!(L, :gap_object) do
-    gap_lie_algebra_by_struct_consts(L)
-  end
+# currently, the user should only use _iso_oscar_gap(LO) to get the isomorphism
```

Comment:
Why?
The idea in `src/GAP/iso_oscar_gap.jl` is that
- `iso_oscar_gap` deals with setting and fetching the attribute value, and calls `_iso_oscar_gap`, and
- the methods for `_iso_oscar_gap` just have to do the work, they need not care about attribute values.

Reply:
I tried to explain that in https://github.com/oscar-system/Oscar.jl/pull/2207#issuecomment-1497783766.

My problem is that, in some instances, I would like to set the iso by myself instead of using the "standard" one. For example, when I use the constructor `lie_algebra(R::Ring, dynkin::Tuple{Char,Int})`, internally `GAP.Globals.SimpleLieAlgebra` gets called, and the result gets translated to Oscar via `iso_gap_oscar`. However, the knowledge about this GAP Lie algebra then gets lost. I would instead prefer to save (the inverse of) this iso into the Oscar Lie algebra here.
This didn't seem possible to me without relying on the internals of AbstractAlgebra on how to name attributes for `@attr`.

If this behavior is not liked or there are some other ideas on how to handle that, comments are very welcome.

Reply:
I just found a naming guarantee of the attribute behind `@attr` in https://github.com/Nemocas/AbstractAlgebra.jl/blob/76a85e28b137ec0b97e2131970a51d443dc3dce7/src/Attributes.jl#L284. With this, I can remove that additional layer of caching and just set the iso myself in the `@attr` storage.

Reply:
This has been addressed in 2e499c4a6c4e92603813d667ef66dfc1828e2a38.